app/dependencies.php

```diff
@@ -420,7 +420,8 @@ static function (ContainerInterface $c): Matching\Adapter {
         StripeClient::class => static function (ContainerInterface $c): StripeClient {
             return new StripeClient([
                 'api_key' => $c->get('settings')['stripe']['apiKey'],
-                'stripe_version' => $c->get('settings')['stripe']['apiVersion'],
+                // Stripe version determined by defaults built into stripe-php library - upgrade the library to get
+                // a new version. See \Stripe\Util\ApiVersion::CURRENT
```

Comment:
Think I agree it makes more sense to couple the version to the code, but I'm not sure about just using the library default. I think this will make it easier to upgrade API without noticing / properly testing, and harder to quickly compare the API use to other apps, vs. setting the version explicitly. I think it's also better that we are able to upgrade to a major SDK version, e.g. to receive a security patch, without committing to testing our entire integration against a new API version on the same timeline.

It's also unlikely we will be able to consistently release changes to all apps that 'talk to' Stripe exactly simultaneously. The env var might be slightly more arguable if we think about this – we will still never get the upgrade out at _exactly_ the same time but we could use it to get a bit closer if we wanted. The question then is whether it's more dangerous to run an untested version in isolation or to knowingly use different versions across the stack simultaneously. IMO it probably depends on what breaking changes we've spotted, so this _might_ be an argument for having the option to redeploy the same app code with a different API version if we preempt problems or observe them after release and need a short notice hotfix to just one of the apps.

There's tension between the options because the env var has the same problem as using the library default in making it harder to just search the code to check what we're using. Overall I think my preferred option is to hardcode and not use the library's default or constant, because the benefits of decoupling dependency upgrades from API ones and quickly seeing what is active both seem more likely to be a real world win than 'hot swapping' MatchBot's API version.

Reply:
From reading the docs it seems like letting the library choose the default is the main recommanded way - https://github.com/stripe/stripe-php/wiki/Migration-guide-for-v12 has "Upgrade your integration to be compatible " as the recommended option, and explicitly specifying an older API version as only something they think you'd want to do to "avoid or postpone upgrading API Versions". They don't suggest at any point that we might want to explicitly specify the current API version.

I don't see much issue with using different API versions across the stack, the normal way of things is that each app should use the API version that it was developed & tested against. I guess the only problem comes when concepts from the API leak into the way the apps communicate between each other - possibly that could be an issue, but its not really any different to one app talking to a later version of itself via stuff saved in a database etc.

And we can easily change the one line of app code that sets the version without changing anything else - for me at least that would be quicker and more reliable than changing an env var since its a routine task. I'd be happy to add something like this to make it a tiny bit easier to edit the version in code:

```
                'stripe_version' => \Stripe\Util\ApiVersion::CURRENT, //  default, set explicitly here for clarity.
```

The changelog for each version of the PHP library has a link to the upgrading docs for the API - e.g. https://github.com/stripe/stripe-php/blob/master/CHANGELOG.md#1500---2024-06-24 links to https://docs.stripe.com/upgrades#2024-06-20 . I feel like it's very good to know that we're using a versions of the the PHP library and API version that were designed to work together.

Reply:
Mm – definitely pro's and con's to each way. If I was designing their API my advice would be the same – but I think the differences in our case all derive from the multiple apps working with the same objects. Simply searching our codebases to quickly check if they're all aligned, and if not by how many versions, is really I think the biggest win.

I do still think decoupling security patches from having to do time consuming testing is also worth thinking about. I'm not sure it's realistic to think we will always remember that a patch comes with this implicit API change, especially if e.g. Dependabot proposed a major version security PR to us. I've also not seen anything suggesting that SDKs aren't designed to support older API versions.

Reply:
As discussed did a combination of hard-coding and relying on library default in fa7077452 by asserting that the two are equal.

Reply:
Hopefully security patches would generally be minor / patch versions of the PHP SDK and not force us to use a new HTTP API version.